controllers/common/common_status_update.go

```diff
@@ -0,0 +1,401 @@
+	if err != nil {
+		return nil, err
+	}
+
+	return decisions, nil
+}
+
+// GetPolicyPlacementDecisions retrieves the placement decisions for a input PlacementBinding when
+// the policy is bound within it. It can return an error if the PlacementBinding is invalid, or if
+// a required lookup fails.
+func GetPolicyPlacementDecisions(ctx context.Context, c client.Client,
+	instance *policiesv1.Policy, pb *policiesv1.PlacementBinding,
+) (decisions []appsv1.PlacementDecision, placements []*policiesv1.Placement, err error) {
+	if !HasValidPlacementRef(pb) {
+		return nil, nil, fmt.Errorf("placement binding %s/%s reference is not valid", pb.Name, pb.Namespace)
+	}
+
+	policySubjectFound := false
+	policySetSubjects := make(map[string]struct{}) // a set, to prevent duplicates
+
+	for _, subject := range pb.Subjects {
+		if subject.APIGroup != policiesv1.SchemeGroupVersion.Group {
+			continue
+		}
+
+		switch subject.Kind {
+		case policiesv1.Kind:
+			if !policySubjectFound && subject.Name == instance.GetName() {
+				policySubjectFound = true
+
+				placements = append(placements, &policiesv1.Placement{
+					PlacementBinding: pb.GetName(),
+				})
+			}
+		case policiesv1.PolicySetKind:
+			if _, exists := policySetSubjects[subject.Name]; !exists {
+				policySetSubjects[subject.Name] = struct{}{}
+
+				if IsPolicyInPolicySet(ctx, c, instance.GetName(), subject.Name, pb.GetNamespace()) {
+					placements = append(placements, &policiesv1.Placement{
+						PlacementBinding: pb.GetName(),
+						PolicySet:        subject.Name,
+					})
+				}
+			}
+		}
+	}
+
+	if len(placements) == 0 {
+		// None of the subjects in the PlacementBinding were relevant to this Policy.
+		return nil, nil, nil
+	}
+
+	// If the placementRef exists, then it needs to be added to the placement item
+	refNN := types.NamespacedName{
+		Namespace: pb.GetNamespace(),
+		Name:      pb.PlacementRef.Name,
+	}
+
+	switch pb.PlacementRef.Kind {
+	case "PlacementRule":
+		plr := &appsv1.PlacementRule{}
+		if err := c.Get(ctx, refNN, plr); err != nil && !k8serrors.IsNotFound(err) {
+			return nil, nil, fmt.Errorf("failed to check for PlacementRule '%v': %w", pb.PlacementRef.Name, err)
+		}
+
+		for i := range placements {
+			placements[i].PlacementRule = plr.Name // will be empty if the PlacementRule was not found
+		}
+	case "Placement":
+		pl := &clusterv1beta1.Placement{}
+		if err := c.Get(ctx, refNN, pl); err != nil && !k8serrors.IsNotFound(err) {
+			return nil, nil, fmt.Errorf("failed to check for Placement '%v': %w", pb.PlacementRef.Name, err)
+		}
+
+		for i := range placements {
+			placements[i].Placement = pl.Name // will be empty if the Placement was not found
+		}
+	}
+
+	// If there are no placements, then the PlacementBinding is not for this Policy.
+	if len(placements) == 0 {
+		return nil, nil, nil
+	}
+
+	// If the policy is disabled, don't return any decisions, so that the policy isn't put on any clusters
+	if instance.Spec.Disabled {
```

Comment:
Is there a reason to not return this in the beginning of the function? Perhaps @JustinKuli remembers.

Reply:
I think there are tests that check the status.placements even when it's disabled?